### PR TITLE
Applied FMMs to build_g.

### PR DESCRIPTION
**Pull request by mnottoli**
build_g now is also accelerated using FMMs for arbitrary multipolar distributions. Since there is no more reason to keep mkrhs, I proceeded to remove it from ddx_driver. It is still used by the tests. Further work may be needed.

### CODE REVIEW

File: src/ddx_driver.f90 [hunk 1]
```diff
@@ -9,200 +9,353 @@
...
+! contributions to the forces.
+if (ddx_data % params % force .eq. 1) then
+    allocate(force(3, ddx_data % params % nsph), stat=info)
+    if (info .ne. 0) then
+        write(6, *) "Allocation failed in ddx_driver"
+        stop 1
+    end if
+end if
 start_time = omp_get_wtime()
-call ddsolve(ddx_data, state, phi_cav, gradphi_cav, hessianphi_cav, psi, &
```
Comment (mfherbst) [inline src/ddx_driver.f90:84]: Doesn't this imply that ddsolve is used nowhere? Maybe we should remove it completely?

**Review by mfherbst — APPROVED** [review]
Fine by me.

File: src/ddx_driver.f90 (hunk 1 above)
Comment (mnottoli) [reply src/ddx_driver.f90:84]: It is still used by the tests. I will remove it entirely then.

Comment (mfherbst) [reply src/ddx_driver.f90:84]: Ok I see. To me the tests don't count as there we are sort of "free to call whatever function we want".

**Pull request merged**